hathor/websocket/protocol.py

```diff
@@ -140,9 +142,14 @@ def fail_if_history_streaming_is_disabled(self) -> bool:
         ))
         return True
 
-    def _create_streamer(self, stream_id: str, search: AddressSearch) -> None:
+    def _create_streamer(self, stream_id: str, search: AddressSearch, window_size: int | None) -> None:
         """Create the streamer and handle its callbacks."""
         self._history_streamer = HistoryStreamer(protocol=self, stream_id=stream_id, search=search)
+        if window_size is not None:
+            if window_size < 0:
+                self._history_streamer.set_sliding_window_size(None)
+            else:
+                self._history_streamer.set_sliding_window_size(window_size)
```

Comment:
question(if-minor): Why not pass the `window_size` on the constructor?

Reply:
I'd rather have a separate call for now because the constructor would have to be able to differentiate between Enabled, Disabled, and Default. I could create a `_NOT_PROVIDED` object to do it but I guess it's not worth the effort.